corehq/apps/data_interfaces/models.py

```diff
@@ -559,11 +560,11 @@ def matches(self, case, now):
 
 
 class MatchPropertyDefinition(CaseRuleCriteriaDefinition):
-    # True when today < (the date in property_name + property_value days)
-    MATCH_DAYS_BEFORE = 'DAYS_BEFORE'
+    MATCH_DAYS_LESS_THAN = 'DAYS_BEFORE'
+    MATCH_DAYS_LESS_OR_EQUAL = 'DAYS_LESS_OR_EQUAL'
 
-    # True when today >= (the date in property_name + property_value days)
-    MATCH_DAYS_AFTER = 'DAYS'
+    MATCH_DAYS_GREATER_THAN = 'GREATER_THAN_DAYS'
```

Comment:
The others all start with `DAYS`
```suggestion
    MATCH_DAYS_GREATER_THAN = 'DAYS_GREATER_THAN'
```

Reply:
Should be fixed now. `_try_date_conversion` apparently returns `datetime` objects when a string is provided, which, unfortunately for us, are also considered instances of `date`. I noticed the two other usages of `_try_date_conversion` in this file have the same logic of forcing the returned `datetime` into a `date` object. I was tempted to just modify the implementation of `_try_date_conversion` to return a `date`, but it looks like there is one other usage, imported by _scheduling/scheduling_partitioned/models_ that specifically wants a `datetime` returned here.  Given that `try_date_conversion` seems to imply that its returning a date anyway, do you think I should modify that function and then just force the `date` into a `datetime` for the _scheduling_partitioned_ code?